platforms/android/library/src/main/java/io/element/android/wysiwyg/viewmodel/EditorViewModel.kt

```diff
@@ -1,36 +1,42 @@
-package io.element.android.wysiwyg.inputhandlers
+package io.element.android.wysiwyg.viewmodel
 
-import android.content.Context
 import android.text.Editable
 import android.text.Spanned
-import androidx.core.text.getSpans
+import androidx.lifecycle.ViewModel
 import io.element.android.wysiwyg.BuildConfig
 import io.element.android.wysiwyg.extensions.log
 import io.element.android.wysiwyg.extensions.string
 import io.element.android.wysiwyg.inputhandlers.models.EditorInputAction
 import io.element.android.wysiwyg.inputhandlers.models.InlineFormat
 import io.element.android.wysiwyg.inputhandlers.models.ReplaceTextResult
-import io.element.android.wysiwyg.spans.ExtraCharacterSpan
 import io.element.android.wysiwyg.utils.EditorIndexMapper
 import io.element.android.wysiwyg.utils.HtmlToSpansParser
-import uniffi.wysiwyg_composer.ComposerModelInterface
+import io.element.android.wysiwyg.utils.ResourcesProvider
+import uniffi.wysiwyg_composer.ComposerModel
 import uniffi.wysiwyg_composer.MenuState
 import uniffi.wysiwyg_composer.TextUpdate
-import kotlin.math.absoluteValue
+import uniffi.wysiwyg_composer.newComposerModel
 
-internal class InputProcessor(
-    private val context: Context,
-    private val menuStateCallback: (MenuState) -> Unit,
-    private val composer: ComposerModelInterface?,
-) {
+internal class EditorViewModel(
+    private val resourcesProvider: ResourcesProvider,
```

Comment:
We'd still be holding a reference to the view context here which could leak it. If we need to do this, I think it could be better to use [`AndroidViewModel`](https://developer.android.com/reference/androidx/lifecycle/AndroidViewModel) that only holds the application context (which there is only one instance of).